### PR TITLE
new rate mode and minor adjustments

### PR DESCRIPTION
**Pull request by RusteyBucket**
Tl;Dr:

I made my own flavor of rate mode called RR for RusteyRate, added a delayer during paused gameplay while on OS time and added a bunch of TODOs I'd ideally like to implement at some point

commit 49b48d19c30a767c6f932b2a1fa2a065327990e8
Merge: cbecf5d 4158208
Author: Rustey <io-01@gmx.ch>
Date:   Tue Jan 25 22:39:28 2022 +0100

    Merge branch 'One' into RusteyRate

    # Conflicts:
    #	res/scripts/cartok/helper.lua

commit 4158208c8af7f52c334200df3fd1e3f553bdca1b
Author: Rustey <io-01@gmx.ch>
Date:   Tue Jan 25 22:00:24 2022 +0100

    Fucked up a variable and prevented paused buying

commit 84754dc26f3d630c66cfe828fc9c2170a6ca74b5
Author: Rustey <io-01@gmx.ch>
Date:   Sun Jan 23 17:22:46 2022 +0100

    need exxaggerated demand for the RR mode

commit 68ef039708dfdc0fe822cf4238daf77914ab3a97
Author: Rustey <io-01@gmx.ch>
Date:   Sun Jan 23 17:05:36 2022 +0100

    no updates during paused and new parameter averageVehicleCapacity because I can't find the frequency

commit 062635f8633509b4d5aa141061ac199e160451f7
Author: Rustey <io-01@gmx.ch>
Date:   Sun Jan 23 13:06:55 2022 +0100

    cleaned up readability of helper.lua/newVehicleConditions

commit b03d6fc794a03559459d51fa28eb9ba5e85c785e
Author: Rustey <io-01@gmx.ch>
Date:   Sun Jan 23 13:01:02 2022 +0100

    adjusted README.md

commit a4b9edb22dc865bfd44849861cba8545326005dc
Author: Rustey <io-01@gmx.ch>
Date:   Sun Jan 23 13:00:16 2022 +0100

    adjusted the factors a bit

commit 97a788b6f4fa002bb07c984658c7d430e1d8ba91
Author: Rustey <io-01@gmx.ch>
Date:   Sat Jan 22 19:20:08 2022 +0100

    added a new RR mode for Rustey's rules and started some other ideas

commit b7df6c7175758a2219ddcee3580709a06acb1067
Author: Rustey <io-01@gmx.ch>
Date:   Fri Jan 21 01:22:27 2022 +0100

    more to the todo list

commit f45a3598ed114f5ba8f24cef55fa18114e326344
Author: Rustey <io-01@gmx.ch>
Date:   Thu Jan 20 22:44:47 2022 +0100

    Tried to get the vehicle price, goddamn, where the fuck is that kind of information?

commit 5f23893b37d91a42adc6dd1676c29efa0d68bce3
Author: Rustey <io-01@gmx.ch>
Date:   Thu Jan 20 18:24:03 2022 +0100

    Added my TODOs and a default setter and mode getter

### CODE REVIEW

**Comment by TommyC81** [comment]
Thanks RusteyBucket, some interesting stuff in there! I'm working on an internal branch that has overhauled many aspects and also introduces CARGO line management. Still needs more refinement before it'll be ready to share. I will incorporate what I can from this PR, please continue working on it. Thanks for your support and assistance!

**Comment by RusteyBucket** [comment]
Oh, great, cargo... I'll see if I can advance my idea of a variable input mode intended for frequency enforcement

**Comment by TommyC81** [comment]
Just a note to let you know I'm making progress. I think I've been running in to performance issues causing updates to take too long, and this causing desyncs and crashes. I've reorganized this extensively and will hopefully have further updates soon - once it's stable.

**Comment by TommyC81** [comment]
Exciting news; after a lot of testing and fixing, I got a stable (as far as I can tell) version that incorporates CARGO management here: https://github.com/TommyC81/TPF2-LineManager/tree/DataRevision
Please give it a go, and push it to its limits!

**Comment by TommyC81** [comment]
Another note to say that I've pushed even more updates to the CARGO update branch (a.k.a DataRevision branch). This includes the availing of frequency as a rule tool. All rules are now contained in a single rules.lua file for (relatively) easy management and changes. Please take a look and let me know.

For re-organization purposes some previous debug functionality has been removed. But nothing that can't be added if needed. Overall, things are much more polished now and seem to work really well.

**Comment by TommyC81** [comment]
I've now pushed the CARGO UPDATE to the live version and updated the main branch. Will close this PR. Feel free to submit a PR targeting the main branch. In particular, check 'rules.lua' in the updated version!
